src/prefect/cli/_prompts.py

```diff
@@ -623,3 +642,178 @@ async def prompt_entrypoint(console: Console) -> str:
+            {"header": "Storage Type", "key": "type"},
+            {"header": "Description", "key": "description"},
+        ],
+        data=valid_storage_options_for_context,
+    )
+
+    return selected_flow_storage_row["slug"]
+
+
+@inject_client
+async def prompt_select_blob_storage_credentials(
+    console: Console, storage_provider: str, client: PrefectClient = None
+) -> str:
+    """
+    Prompt the user for blob storage credentials.
+
+    Returns a jinja template string that references a credentials block.
+    """
+
+    storage_provider_slug = storage_provider.replace("_", "-")
+    pretty_storage_provider = storage_provider.replace("_", " ").upper()
+
+    creds_block_type_slug = STORAGE_PROVIDER_TO_CREDS_BLOCK[storage_provider]
+    pretty_creds_block_type = creds_block_type_slug.replace("-", " ").title()
+
+    existing_credentials_blocks = await client.read_block_documents_by_type(
+        block_type_slug=creds_block_type_slug
+    )
+
+    if existing_credentials_blocks:
+        selected_credentials_block = prompt_select_from_table(
+            console,
+            prompt=(
+                f"Select from your existing {pretty_creds_block_type} credential blocks"
+            ),
+            columns=[
+                {
+                    "header": f"{pretty_storage_provider} Credentials Blocks",
+                    "key": "name",
+                }
+            ],
+            data=[{"name": block.name} for block in existing_credentials_blocks],
+            opt_out_message="Create a new credentials block",
+        )
+
+        if selected_credentials_block and (
+            selected_block := selected_credentials_block.get("name")
+        ):
+            return f"{{{{ prefect.blocks.{creds_block_type_slug}.{selected_block} }}}}"
+
+    credentials_block_type = await client.read_block_type_by_slug(creds_block_type_slug)
+
+    credentials_block_schemas = (
+        await client._client.post(
+            "/block_schemas/filter",
+            json={
+                "block_schemas": {
+                    "block_type_id": {"any_": [str(credentials_block_type.id)]}
+                },
+                "limit": 1,
+            },
+        )
+    ).json()
+
+    credentials_block_schema = parse_obj_as(BlockSchema, credentials_block_schemas[0])
+
+    console.print(
+        f"\nProvide details on your new {pretty_storage_provider} credentials:"
+    )
+
+    hydrated_fields = {
+        field_name: prompt(f"{field_name} [yellow]({props.get('type')})[/]")
+        for field_name, props in credentials_block_schema.fields.get(
+            "properties"
+        ).items()
+        if field_name in REQUIRED_FIELDS_FOR_CREDS_BLOCK[creds_block_type_slug]
+    }
+
+    console.print(f"[blue]\n{pretty_storage_provider} credentials specified![/]\n")
+
+    credentials_block_name = prompt(
+        "Give a name to your new credentials block",
+        default=f"{storage_provider_slug}-storage-credentials",
+    )
```

Comment:
It might be worthwhile to build the 'name already exists' validation into a custom prompt so `rich` can handle prompting as many times as necessary.